### PR TITLE
Switch string_formatter to use fmtlib

### PR DESCRIPTION
**Pull request by olanti-p**
Switch `string_formatter` to use `{fmt}` library https://github.com/fmtlib/fmt .
Clean up tests a bit (includes cherry-picked https://github.com/CleverRaven/Cataclysm-DDA/commit/a24465f2ea21a114338f5e21e5a479a511b7e1d5 ).
Should fix failing tests on Windows, and make `string_formatter` more robust.

Some pain points:
1. fmtlib adds signs to hex/oct numbers when formatting with `+` or `(space)` flags
    * added a workaround for that
2. fmtlib does not suport positional width/precision specifiers
    * `string_formatter` actually takes care of that, so not a problem here
3. fmtlib outputs "0" instead of an empty string when formatting zero integers with zero width
    * seems harmless to me, even if it goes against `printf` specs. Changed tests to match fmtlib behavior

Also, it appears that the are no formatting strings in the codebase that fall under any of these 3 cases.
I tried searching with ripgrep
1. `rg "%[0-9\$#* +'I-]*[oxX]" --sort path`
2. `rg "\*[0-9]+\\$" --sort path`
3. `rg "%[0-9\$#* +'I-]*\.[\$*0-9lhzt]*[oxXudi]" --sort path`

and found no matches besides the ones from `string_formatter_test.cpp`.

TODO:
- [x] deal with remaining `printf` calls while avoiding including `printf.h` everywhere
- [x] add `fmtlib_` prefix to fmtlib files' names
- [x] some cleanup (rename stuff, remove list of failing tests, squash some commits, etc.)
- [ ] ~search for an alternative solution?~

### CODE REVIEW

**Comment by Coolthulhu** [comment]
Tiny note: the license of `{fmt}` requires including its license somewhere in the repo, though not necessarily in binaries.

**Comment by olanti-p** [comment]
Would https://github.com/cataclysmbnteam/Cataclysm-BN/blob/upload/LICENSE.txt do?

**Comment by Coolthulhu** [comment]
I think it would, it doesn't say "include the file", just "include the notice".

File: tests/vehicle_efficiency_test.cpp [hunk 1]
```diff
@@ -310,25 +311,19 @@ static int average_from_stat( const efficiency_stat &st )
...
-    printf( "%s", ss.str().c_str() );
-    fflush( stdout );
+    int v1 = avg_from_stat( find_inner( type, expected_mass, "t_pavement", -1, false, false ) );
+    int expm = expected_mass;
+    int v2 = avg_from_stat( find_inner( type, expected_mass, "t_dirt", -1, false, false ) );
+    int v3 = avg_from_stat( find_inner( type, expected_mass, "t_pavement", 5, false, false ) );
+    int v4 = avg_from_stat( find_inner( type, expected_mass, "t_dirt", 5, false, false ) );
+
+    cata_printf(
+        "    test_vehicle( \"%s\", %d; %d, %d, %d, %d );",
```
Comment (olanti-p) [inline tests/vehicle_efficiency_test.cpp:324]:
```suggestion
        "    test_vehicle( \"%s\", %d; %d, %d, %d, %d );\n",
```

**Comment by Coolthulhu** [comment]
Fixed the dreaded build-stopper. Great work!

**Pull request merged**